util-cache/src/main/scala/sbt/util/ActionCache.scala

```diff
@@ -37,30 +37,47 @@ object ActionCache:
   )(
       config: BuildWideCacheConfiguration
   ): O =
-    val store = config.store
-    val cacheEventLog = config.cacheEventLog
+    import config.*
     val input =
       Digest.sha256Hash(codeContentHash, extraHash, Digest.dummy(Hasher.hashUnsafe[I](key)))
-    val valuePath = config.outputDirectory.resolve(s"value/${input}.json").toString
+    val valuePath = s"value/${input}.json"
+
     def organicTask: O =
-      cacheEventLog.append(ActionCacheEvent.NotFound)
       // run action(...) and combine the newResult with outputs
-      val (newResult, outputs) = action(key)
-      val json = Converter.toJsonUnsafe(newResult)
-      val valueFile = StringVirtualFile1(valuePath, CompactPrinter(json))
-      val newOutputs = Vector(valueFile) ++ outputs.toVector
-      store.put(UpdateActionResultRequest(input, newOutputs, exitCode = 0)) match
-        case Right(result) =>
-          store.syncBlobs(result.outputFiles, config.outputDirectory)
-          newResult
-        case Left(e) => throw e
+      val (result, outputs) =
+        try action(key)
+        catch
+          case e: Exception =>
+            cacheEventLog.append(ActionCacheEvent.Error)
+            throw e
+      val json = Converter.toJsonUnsafe(result)
+      val uncacheableOutputs =
+        outputs.map(fileConverter.toPath).filter(f => !f.startsWith(outputDirectory))
+      if uncacheableOutputs.nonEmpty then
+        cacheEventLog.append(ActionCacheEvent.Error)
+        logger.error(
+          s"Cannot cache task because its output files do not start with $outputDirectory: \n" +
+            uncacheableOutputs.mkString("  - ", "\n  - ", "")
+        )
```

Comment:
This is more cosmetic thing, but I personally don't like build tools leaking out the absolute paths in the log messages, since it could include information like name or company that the build user may not want to publicize. Some of the existing messages include them, but we should avoid that for new entries esp. remote cache.

Reply:
Updated message is:
```
sbt:sbt2-test> compile
[error] Cannot cache task because its output files are outside the output directory: 
[error]   - ${BASE}/custom-target/zinc/inc_compile_3.zip
[error]   - ${BASE}/custom-target/sbt2-test_3-0.1.0-SNAPSHOT-noresources.jar
[success] elapsed time: 0 s, cache 0%, 1 error
```